components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerV2.java

```diff
@@ -0,0 +1,1138 @@
+        if (permissions != null) {
+            for (Permission permission : permissions) {
+                MultiValuedComplexType permissionComplexObject = new MultiValuedComplexType();
+                permissionComplexObject.setValue(permission.getName());
+                permissionComplexObject.setDisplay(permission.getDisplayName());
+                permissionValues.add(permissionComplexObject);
+            }
+        }
+        return permissionValues;
+    }
+
+    public void deleteRole(String roleID) throws CharonException, NotFoundException, BadRequestException {
+
+        try {
+            if (!isRoleModificationAllowedForTenant(tenantDomain)) {
+                throw new BadRequestException("Role deletion is not allowed for organizations.",
+                        ResponseCodeConstants.INVALID_VALUE);
+            }
+            roleManagementService.deleteRole(roleID, tenantDomain);
+        } catch (IdentityRoleManagementException e) {
+            if (StringUtils.equals(ROLE_NOT_FOUND.getCode(), e.getErrorCode())) {
+                throw new NotFoundException(e.getMessage());
+            } else if (StringUtils.equals(OPERATION_FORBIDDEN.getCode(), e.getErrorCode())) {
+                throw new BadRequestException(e.getMessage());
+            }
+            throw new CharonException(String.format("Error occurred while deleting the role: %s", roleID), e);
+        }
+    }
+
+    public RolesV2GetResponse listRolesWithGET(Node rootNode, Integer startIndex, Integer count, String sortBy,
+                                               String sortOrder)
+            throws CharonException, NotImplementedException, BadRequestException {
+
+        if (sortBy != null || sortOrder != null) {
+            throw new NotImplementedException("Sorting is not supported.");
+        } else if (count != null && count == 0) {
+            return new RolesV2GetResponse(0, Collections.emptyList());
+        } else if (rootNode != null) {
+            return filterRoles(rootNode, count, startIndex, null, null);
+        } else {
+            return listRoles(count, startIndex, null, null);
+        }
+    }
+
+    @Override
+    public RolesV2GetResponse listRolesWithPost(SearchRequest searchRequest)
+            throws NotImplementedException, BadRequestException, CharonException {
+
+        return listRolesWithGET(searchRequest.getFilter(), searchRequest.getStartIndex(), searchRequest.getCount(),
+                searchRequest.getSortBy(), searchRequest.getSortOder());
+    }
+
+    @Override
+    public RoleV2 updateRole(RoleV2 oldRole, RoleV2 newRole)
+            throws BadRequestException, CharonException, ConflictException, NotFoundException {
+
+        doUpdateRoleName(oldRole, newRole);
+        doUpdateUsers(oldRole, newRole);
+        doUpdateGroups(oldRole, newRole);
+        doUpdatePermissions(oldRole, newRole);
+
+        RoleV2 role = new RoleV2();
```

Comment:
Introducing a builder for RoleV2 may helpful

Reply:
RoleV2  is an object of Charon, Since the setters of this class is not just attribute setters we need to duplicate them in builder class as well. Hence keeping as it is